diff --git a/Hedgehog/Base/Type/detail/hhStringHolder.h b/Hedgehog/Base/Type/detail/hhStringHolder.h
index bc3dbc2..5ee8428 100644
--- a/Hedgehog/Base/Type/detail/hhStringHolder.h
+++ b/Hedgehog/Base/Type/detail/hhStringHolder.h
@@ -10,7 +10,7 @@ namespace Hedgehog::Base
         char aStr[];

         static SStringHolder* GetHolder(const char* in_pStr);
-        static SStringHolder* Make(const char* in_pStr);
+        static SStringHolder* Make(const char* in_pStr, size_t length);

         void AddRef();
         void Release();
diff --git a/Hedgehog/Base/Type/detail/hhStringHolder.inl b/Hedgehog/Base/Type/detail/hhStringHolder.inl
index 0daf6e6..1256adf 100644
--- a/Hedgehog/Base/Type/detail/hhStringHolder.inl
+++ b/Hedgehog/Base/Type/detail/hhStringHolder.inl
@@ -1,3 +1,7 @@
+#include <cstring>
+#include <cstdint>
+#include <atomic>
+
 namespace Hedgehog::Base
 {
     inline SStringHolder* SStringHolder::GetHolder(const char* in_pStr)
@@ -5,11 +9,22 @@ namespace Hedgehog::Base
         return (SStringHolder*)((size_t)in_pStr - sizeof(RefCount));
     }

-    inline SStringHolder* SStringHolder::Make(const char* in_pStr)
+    inline SStringHolder* SStringHolder::Make(const char* in_pStr, size_t length)
     {
-        auto pHolder = (SStringHolder*)__HH_ALLOC(sizeof(RefCount) + strlen(in_pStr) + 1);
-        pHolder->RefCount = 1;
-        strcpy(pHolder->aStr, in_pStr);
+        constexpr uint32_t overhead = sizeof(RefCount) + 1;
+
+        if (length > (uint32_t)-1 - overhead)
+            return nullptr;
+
+        auto pHolder = (SStringHolder*)__HH_ALLOC(overhead + (uint32_t)length);
+
+        if (pHolder)
+        {
+            pHolder->RefCount = 1;
+            std::memcpy(pHolder->aStr, in_pStr, length);
+            pHolder->aStr[length] = '\0';
+        }
+
         return pHolder;
     }

diff --git a/Hedgehog/Base/Type/hhSharedString.inl b/Hedgehog/Base/Type/hhSharedString.inl
index 2bddc71..6f04784 100644
--- a/Hedgehog/Base/Type/hhSharedString.inl
+++ b/Hedgehog/Base/Type/hhSharedString.inl
@@ -1,3 +1,5 @@
+#include <cstring>
+
 namespace Hedgehog::Base
 {
     inline SStringHolder* CSharedString::GetHolder() const
@@ -17,8 +19,11 @@ namespace Hedgehog::Base
     {
         size_t length;

-        if (in_pStr && (length = strlen(in_pStr)) != 0)
-            m_pStr.ptr = g_memory.MapVirtual(SStringHolder::Make(in_pStr)->aStr);
+        if (in_pStr && (length = std::strlen(in_pStr)) != 0)
+        {
+            if (auto* pHolder = SStringHolder::Make(in_pStr, length))
+                m_pStr.ptr = g_memory.MapVirtual(pHolder->aStr);
+        }
     }

     inline CSharedString::CSharedString(const CSharedString& in_rOther) : m_pStr(in_rOther.m_pStr)
@@ -53,7 +58,7 @@ namespace Hedgehog::Base

     inline size_t CSharedString::size() const
     {
-        return strlen(m_pStr);
+        return std::strlen(m_pStr);
     }

     inline size_t CSharedString::length() const
